### PR TITLE
sema: some API fixes

### PR DESCRIPTION
**Pull request by authmillenon**
* fixes (and documents properly) return values of `sema_post()`
* introduces static initializer for semaphores
* some style fixes.

### CODE REVIEW

File: sys/include/sema.h [hunk 1]
```diff
@@ -31,6 +31,15 @@ extern "C" {
 #endif
 
 /**
+ * @brief   Creates semaphore statically.
+ *
+ * @param[in] value Initial value for the semaphore.
+ *
+ * @return  Statically initialized semaphore.
+ */
+#define SEMA_CREATE(value)      { value, PRIORITY_QUEUE_INIT }
```
Comment (Kijewski) [inline sys/include/sema.h]: `(value)`

Comment (authmillenon) [inline sys/include/sema.h]: Fixed

**Comment by Kijewski** [comment]
Looks good, [`int sem_post(sem_t*)`](https://github.com/RIOT-OS/RIOT/blob/7af2a62b123202139cfbe67a11490b708aa7735e/sys/posix/include/semaphore.h#L127) should not have worked correctly before.

**Comment by authmillenon** [comment]
Was that an ACK, @Kijewski?

**Comment by authmillenon** [comment]
Squashed and rebased to current master

**Comment by josephnoir** [comment]
Why does the semaphore use a volatile int instead of an atomic int, like the [mutex implementation](https://github.com/RIOT-OS/RIOT/blob/master/core/include/mutex.h) does?

**Comment by authmillenon** [comment]
Because the core of this implementation is older than the current iteration of `core_atomic`. Will, but in another PR, since this is not related to the topic of this PR.

**Comment by jfischer-phytec-iot** [comment]
ACK from my side

**Pull request merged**